components/calendar/src/any_calendar.rs

```diff
@@ -858,164 +851,6 @@ pub enum AnyCalendarKind {
 }
 
 impl AnyCalendarKind {
-    /// Construct from a BCP-47 string
-    ///
-    /// Returns `None` if the calendar is unknown.
-    pub fn get_for_bcp47_string(x: &str) -> Option<Self> {
-        Self::get_for_bcp47_bytes(x.as_bytes())
-    }
-    /// Construct from a BCP-47 byte string
-    ///
-    /// Returns `None` if the calendar is unknown.
-    pub fn get_for_bcp47_bytes(x: &[u8]) -> Option<Self> {
```

Comment:
question: is there a valid alternative to use besides `get_for_bcp47_bytes` and `as_bcp47_string`?

`temporal_rs` uses these methods downstream, but maybe there's a better approach to use

Reply:
Yes, the API on [`CalendarAlgorithm`](https://unicode-org.github.io/icu4x/rustdoc/icu/calendar/preferences/enum.CalendarAlgorithm.html). That actually complies with BCP-47, unlike this which fails on `islamic-rgsa`, and accepts `japanext`.

Reply:
@zbraniecki do preference support aliases?